aip/client-libraries/4222.md

```diff
@@ -154,6 +170,54 @@ If none of the routing parameters matched their respective fields, the routing h
 Much like URL parameters, if there is more than one key-value pair to be sent, the `&`
 character is used as the separator.
 
+### `path_template` syntax
+
+As seen in the above examples, the `path_template` can use a variety of symbols
+that are interpreted by code generators during conversion to regular expressions
+or non-regular expression matcher implementations. The `path_template` consists
+of segments delimited by the segment delimiter. The syntax for `path_template`
+is as follows:
+
+- The only acceptable segment delimiter is `/`.
+  - The last symbol in a `path_template` **may** be a delimiter - it will be 
+    ignored.
+- A segment **must** be of one of the following types:
+  - `*`: A single-segment wildcard. Corresponds to 1 or more non-`/` symbols.
+    The regex describing it is `[^\/]+`.
+    - A Single-segment wildcard typically represents a resource ID.
+  - `**`: A multi-segment wildcard. Corresponds to 0 or more segments. 
+    - A multi-segment wildcard **must** only appear as the final segment or
+      make up the entire `path_template`.
+    - In a multi-segment `path_template`, a multi-segment wildcard **must**
+      appear immediately following a segment delimiter. This delimiter is
+      consumed while matching so a `path_template` like `foo/**` matches all of
+      the following: `foo`, `foo/`, `foo/bar/baz`.
+    - In a multi-segment `path_template`, when used as the last segment the
+      regex describing it is `([:\/].*)?`.
```

Comment:
same re: backslash

also, why colon - is it a special character for path templates in this context?

Reply:
Yes, `:` is used in custom methods like `/v1/{name=books/*}:sort`